### PR TITLE
Added documentation about Julia.

### PR DESCRIPTION
**Pull request by sjsillan**

### CODE REVIEW

**Comment by Nortamo** [comment]
The application needs to be added to the software listing.

File: docs/apps/julia.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+function library.
+
+For a quick introduction and tutorial, see https://github.com/csc-training/julia-introduction 
+
+[TOC]
+
+## Available
+
+### Puhti
+-   1.1.1 compiled with Intel Math Kernel Library (MKL)
```
Comment (Nortamo) [inline docs/apps/julia.md]: The file appl/soft/math/julia/gcc-mkl/julia-1.1.1/usr has the wrong permissions, and is not accessible

File: docs/apps/julia.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+-   1.1.1 from pre-compiled binaries
```
Comment (Nortamo) [inline docs/apps/julia.md]: Some more information for the users could be needed, i.e when should you use which version?

Comment (sjsillan) [reply docs/apps/julia.md]: I added recommendation to use pre-compiled binaries, when using ARPACK package.

**Pull request merged**